### PR TITLE
fastjet: jet clustering using delaunay first stab

### PR DESCRIPTION
**Pull request by Bastiantheone**
PR to start the implementation of jet clustering using the Delaunay triangulation.

### CODE REVIEW

File: fastjet/clustersequence.go [hunk 1]
```diff
@@ -498,3 +499,44 @@ func (cs *ClusterSequence) runN3Dumb() error {
...
+// with coincident eta-phi coordinates.
+func (cs *ClusterSequence) runNlnN() error {
+	panic(fmt.Errorf("fastjet: runNlnN not implemented"))
+}
+
+// addKtDistance adds the current kt distance for particle jeti to the heap
+// using information about the nearest neighbor in the eta-phi plane.
+// Work as follows:
+//
+// . if the kt is zero then it's nearest neighbour is taken to be the
```
Comment (sbinet) [inline fastjet/clustersequence.go]: `s/it's/its/`

File: fastjet/clustersequence.go [hunk 2]
```diff
@@ -498,3 +499,44 @@ func (cs *ClusterSequence) runN3Dumb() error {
...
+//   the beam jet and the distance is zero.
```
Comment (sbinet) [inline fastjet/clustersequence.go]: `s/to be the the beam/to be the beam/`

File: fastjet/clustersequence.go [hunk 3]
```diff
@@ -498,3 +499,44 @@ func (cs *ClusterSequence) runN3Dumb() error {
...
+//
+// . if cylinder distance to nearest neighbour > _Rparam then it is
+//   yiB that is smallest and this is added to map.
```
Comment (sbinet) [inline fastjet/clustersequence.go]: `s/to map/to the heap/` ? right?

File: fastjet/clustersequence.go [hunk 4]
```diff
@@ -498,3 +499,44 @@ func (cs *ClusterSequence) runN3Dumb() error {
...
+//
+// . otherwise if the nearest neighbour jj has a larger kt then add
+//   dij to the map.
```
Comment (sbinet) [inline fastjet/clustersequence.go]: `s/to the map/to the heap/` ?

File: fastjet/clustersequence.go [hunk 5]
```diff
@@ -498,3 +499,44 @@ func (cs *ClusterSequence) runN3Dumb() error {
...
+}
+
+// addKtDistance adds the current kt distance for particle jeti to the heap
+// using information about the nearest neighbor in the eta-phi plane.
+// Work as follows:
+//
+// . if the kt is zero then it's nearest neighbour is taken to be the
+//   the beam jet and the distance is zero.
+//
+// . if cylinder distance to nearest neighbour > _Rparam then it is
```
Comment (sbinet) [inline fastjet/clustersequence.go]: `_Rparam`... this looks like a C++ private member :)

Comment (Bastiantheone) [reply fastjet/clustersequence.go]: Oops :)

Comment (Bastiantheone) [reply fastjet/clustersequence.go]: Would that be r^2 in our case?

File: fastjet/clustersequence.go (hunk 1 above)
Comment (Bastiantheone) [reply fastjet/clustersequence.go]: done

File: fastjet/clustersequence.go (hunk 2 above)
Comment (Bastiantheone) [reply fastjet/clustersequence.go]: done

File: fastjet/clustersequence.go (hunk 3 above)
Comment (Bastiantheone) [reply fastjet/clustersequence.go]: done

File: fastjet/clustersequence.go (hunk 4 above)
Comment (Bastiantheone) [reply fastjet/clustersequence.go]: done

File: fastjet/clustersequence.go (hunk 5 above)
Comment (sbinet) [reply fastjet/clustersequence.go]: yep.

**Comment by Bastiantheone** [comment]
Done

**Pull request merged**